src/giftCards/components/GiftCardExpirySelect/GiftCardExpirySelect.tsx

```diff
@@ -0,0 +1,135 @@
+import { TextField, Typography } from "@material-ui/core";
+import HorizontalSpacer from "@saleor/apps/components/HorizontalSpacer";
+import VerticalSpacer from "@saleor/apps/components/VerticalSpacer";
+import ControlledCheckbox from "@saleor/components/ControlledCheckbox";
+import RadioGroupField from "@saleor/components/RadioGroupField";
+import { GiftCardError } from "@saleor/fragments/types/GiftCardError";
+import { GiftCardExpiryType } from "@saleor/giftCards/GiftCardCreateDialog/types";
+import { getExpiryPeriodTerminationDate } from "@saleor/giftCards/GiftCardCreateDialog/utils";
+import { getGiftCardErrorMessage } from "@saleor/giftCards/GiftCardUpdate/messages";
+import { FormChange } from "@saleor/hooks/useForm";
+import { TimePeriodTypeEnum } from "@saleor/types/globalTypes";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+import { MessageDescriptor, useIntl } from "react-intl";
+
+import TimePeriodField from "../TimePeriodField/TimePeriodField";
+import { giftCardExpirySelectMessages as messages } from "./messages";
+import { useGiftCardExpirySelectStyles as useStyles } from "./styles";
+
+interface UntranslatedOption {
+  label: MessageDescriptor;
+  value: GiftCardExpiryType;
+}
+
+const options: UntranslatedOption[] = [
+  {
+    label: messages.expiryPeriodLabel,
+    value: "EXPIRY_PERIOD"
+  },
+  {
+    label: messages.expiryDateLabel,
+    value: "EXPIRY_DATE"
+  }
+];
+
+interface GiftCardExpirySelectProps {
+  change: FormChange;
+  expirySelected: boolean;
+  expiryPeriodType: TimePeriodTypeEnum;
+  expiryPeriodAmount: number;
+  expiryType: GiftCardExpiryType;
+  customOptions?: UntranslatedOption[];
+  errors?: Record<"expiryDate", GiftCardError>;
+  expiryDate?: string;
+}
+
+const GiftCardExpirySelect: React.FC<GiftCardExpirySelectProps> = ({
```

Comment:
About this component - we have 3 places now where we do this sort of selection:
- gift card settings
<img width="344" alt="Screenshot 2021-09-06 at 13 35 26" src="https://user-images.githubusercontent.com/28022710/132211509-fb85af97-f350-43f4-aeab-92bd1582d280.png">

- gift card update 
<img width="406" alt="Screenshot 2021-09-06 at 13 35 37" src="https://user-images.githubusercontent.com/28022710/132211556-ecc7f099-b49c-45b6-96d5-b35c23d1bea7.png">

- gift card create
<img width="377" alt="Screenshot 2021-09-06 at 13 35 45" src="https://user-images.githubusercontent.com/28022710/132211581-83b8ed0c-9405-4ff6-9ed8-a946cda9c16a.png">

As we can see on the screens, they're all pretty much different, regarding strings, substrings, custom content (like the displayed date in create dialog) options to show in each one etc. That's why I'd rather keep them separate components instead of generic components with custom options etc.

TLDR; please change this component (logic, name and placement in the dir) to GiftCardCreateExpirySelect, remove the custom options etc. If needed, also make one for GiftCardSettingsExpirySelect